### DISCUSSION TITLE
Non-critical error 'chat_id is none' after update

### DISCUSSION THREAD

**Discussion by laroy-sh** [swar/Swar-Chia-Plot-Manager#634] [Q&A] [answered]
Hi Swar, thank you for this app so much, it's fantastic, I swear I'll donate as soon as I get my reward!
I've just updated the app and I can see some non-critical error thrown during start:
Couldn't get chat_id!
         list index out of range
chat_id is none, nothing sent!
Plot Manager has started...

I guess it may have something to do with the Telegram notification (thank you for adding it!!!) that I've configured with a token, but what is a chat_id? There's no such setting in the config.

I'm still not sure if the notification is working or not.
By the way, I only saw notifications about manager start and plot ready, but never about manager stopped or disk full, which are IMHO the most important. Are there any plans to let us configure the notifications?

**Comment by alx741** [comment, marked as answer]
make sure to surround your token with singles quotes like (replace with the token you've got from bot_father):

```yaml
  notify_telegram: true
  telegram_token: '688*****888:AA****nbJW5p*****40T21******v7I*****E'
```

Then, **before** starting the manager, send a Telegram message to your bot, you can do that by searching for it with the name you gave it, then send any message to it ("hi" is fine). Now you can start the manager `$ python manager.py start` and it should work.

**Reply by k0b1** [reply]
discord is working fine, but telegram no :/

**Reply by alx741** [reply]
@k0b1 have you tried this solution yet?

**Reply by galaxy-god** [reply]
@alx741 Hey! I've tried replaced a token from bot_father, save the config file and run the manager.py start. did everything you said above, but still didn't get any  notification in telegram. any help? thanks!

**Reply by lipefur** [reply]
`Couldn't get chat_id!
         list index out of range`

Even putting as quotes I have this error

**Reply by alx741** [reply]
@galaxy-god after running `manager.py start`, did it complain about `Couldn't get chat_id`? or did it just run without printing out an error?

**Reply by galaxy-god** [reply]
@alx741 It just run without an error like normally.

**Reply by galaxy-god** [reply]
Now Im in with the `Couldn't get chat_id! list index out of range ` squad haha

**Reply by dieg0nob** [reply]
I can't get the notifications in telegram and pushover to work, I'm using windows 10 pro
I have put in the config.yaml log_level: INFO to see what was happening, and no error appears.
I already had a telegram bot running with other programs, I have tried to put the key with quotes and without them, without lucky 
I have tried to create a new bot thinking that it could be a telegram problem and I had the same result, nothing arrives and the log does not you see nothing.
Could it be a windows 10 issue? where could i should check?
I have already installed the notification requirements.

**Reply by alx741** [reply]
@dieg0nob have you tried sending a message (via Telegram) to your bot?,  even if you've done that in the past do it again now, just say "hi" to it and restart the manager

**Reply by jabreu-soap** [reply]
I did everything and nothing works.
- Install Dependencies
- Created another bot
- Putted the token in config file with simple quotes
- Stopped and Restarted
- Talked with the bot

**Reply by jabreu-soap** [reply]
`Couldn't get chat_id! list index out of range`

**Reply by alx741** [reply]
@jocieldo-soap "Talked with the bot", but do that right *before* restarting the manager

**Reply by dieg0nob** [reply]
> @dieg0nob have you tried sending a message (via Telegram) to your bot?, even if you've done that in the past do it again now, just say "hi" to it and restart the manager

Yes, I have tried sending a message to the bot, and restarting the manager many times, approximately 15 times, and without any result and suddenly when I finished a plot, I received a notification to the bot, I cannot explain why it did not work, surely it is some telegram stuff. Now if i restart or finish a plot, receive a msj.
Thanks!

**Reply by cherymann** [reply]
There is a new pull request In the pipeline that adds the option to insert your chat_id that will fix this problem. #757

**Reply by kissaev** [reply]
![image](https://user-images.githubusercontent.com/5376567/119391552-259d7a00-bcd7-11eb-9301-b6a0569cc819.png)
I just hardcoded my id in Swar-Chia-Plot-Manager\venv\Lib\site-packages\telegram_notifier\telegram_notifier.py and notifications works fine

**Comment by alx741** [comment]
@lipefur Try this:

0. Make sure you ran `pip isntall -r requirements-notification.txt`
1. run `python manager.py stop`
2. Make double sure the token you have in the config is what the *BotFather* gave you
3. Text your bot again (say "hi" to it), and make sure it is actually your bot (it's easy to text another user/bot by mistake instead of your own if the name is similar)
4.  Run `python manager.py start`
5. See if it works this time around

**Reply by k0b1** [reply]
Everything is cool but to make it works for me I have had to add chat_id: number inside config.
you can get easily that number by paste your bot token number https://q3qkk.csb.app/
in that output search for "group" word and in my case it was "supergroup" = -number
Now its works :)
ps make sure that you start chat with bot so chat id will be created:

**Reply by kissaev** [reply]
> Everything is cool but to make it works for me I have had to add chat_id: number inside config.

Where do you add chat_id, in config.yaml file near bot token?

**Reply by alx741** [reply]
There is no way to add the `chat_id` in the config file... My guess is you made it work and have a phony "chat_id" setting that is doing nothing, but you believe it does.

**Reply by k0b1** [reply]
Ok that looks like this has second side ( and I have experienced something new as this error has pop-up on my screen) for my setup after python manager.py start fallowing your howto I got error that chat id is missing.

so I have input myself new line to confing  - that's help me and solved my problem.

# TELEGRAM
  notify_telegram: true
  chat_id: -XXXXXyour number ( this '-' appear only for my chat_id number - yours might be without...)
  telegram_token: 'your token for bot from botfather'

but for purpose of this conversation I have attempt to #chat_id number to see if error will appear once more...
the answer is - no error anymore.

conclusion: chat_id error might appear.... u can solve it by adding chat_id number and from that moment its become useless :) as bot will know where is located your group :)

**Reply by kissaev** [reply]
> There is no way to add the `chat_id` in the config file... My guess is you made it work and have a phony "chat_id" setting that is doing nothing, but you believe it does.

yea, notifications working fine without "chat_id", thanks
...
I got excited early. It stop working, and i got this error too `Couldn't get chat_id! list index out of range`

**Reply by cherymann** [reply]
#757 is coming with the chat_id in the config file

**Comment by CheepingItReal** [comment]
For me the issue was that I didn't do:

`pip install -r requirements-notification.txt`

**Reply by alx741** [reply]
Oh yeah! don't forget that folks! good catch

**Comment by alx741** [comment]
@galaxy-god

Make sure you ran `pip isntall -r requirements-notification.txt`, then restart the manager.

If that doesn't work then it might be that the configuration of your bot (on the Telegram side of things) got broken somehow. The *BotFather* lets you delete a bot (text him for instructions on that), try and delete yours and then start over creating a new one, you'll get a new token, put that one in the `config.yaml`, make sure to text "hi" to the *new* bot, and then start the manager.

**Reply by galaxy-god** [reply]
@alx741 Oh yeah, I completely forgot about that, Can I run it while my plots are running? Thanks!

**Reply by alx741** [reply]
I believe so... no promises though

**Reply by galaxy-god** [reply]
@alx741 Thanks man, much appreciated!

**Comment by brux88** [comment]
hello, congratulations for the wonderful project, but if I created my bot and put it inside a group, how does it open the chat id of that group, is it possible?
